numba/npyufunc/_internal.c

```diff
@@ -104,6 +105,557 @@ PyTypeObject PyUFuncCleaner_Type = {
     0,                                          /* tp_version_tag */
 };
 
+/* ______________________________________________________________________
+ * DUFunc: A call-time (hence dynamic) specializable ufunc.
+ */
+
+typedef struct {
+    PyObject_HEAD
+    PyObject      * dispatcher;
+    PyUFuncObject * ufunc;
+    PyObject      * keepalive;
+    int             frozen;
```

Comment:
The frozen flag will be set either when the user is satisfied with the loops registered with the DUFunc, or when vectorize is modified to return DUFunc's when signatures are given at decoration-time.